src/index.ts

```diff
@@ -82,14 +82,26 @@ export class KeyboardNavigation {
     workspace.getParentSvg().setAttribute('tabindex', '-1');
 
     this.focusListener = () => {
-      this.navigationController.setHasFocus(workspace, true);
+      this.navigationController.updateWorkspaceFocus(workspace, true);
     };
     this.blurListener = () => {
-      this.navigationController.setHasFocus(workspace, false);
+      this.navigationController.updateWorkspaceFocus(workspace, false);
     };
 
     workspace.getSvgGroup().addEventListener('focus', this.focusListener);
     workspace.getSvgGroup().addEventListener('blur', this.blurListener);
+
+    const toolbox = workspace.getToolbox();
+    if (toolbox != null && toolbox instanceof Blockly.Toolbox) {
+      const contentsDiv = toolbox.HtmlDiv?.querySelector('.blocklyToolboxContents');
+      contentsDiv?.addEventListener('focus', () => {
```

Comment:
You've tried to generalize this to handle either toolbox or flyout, but does this selector actually work in the case where there is a flyout with no toolbox?

We're only testing with category toolboxes so I'm also fine with this code just setting focus on the toolbox and not trying to handle flyouts--would rather have that instead of generalized code that doesn't work in one of two cases.

Reply:
I was mostly aiming to mimic existing logic around how the toolbox was handled elsewhere, but no I haven't tested the case of a flyout with no toolbox. I think it makes sense to simplify the solution here to keep it specific to toolboxes.